### ISSUE TITLE
Comparison to TupleVectors and TypedTables?

### ISSUE THREAD

**Issue by cscherrer** [tkf/RecordArrays.jl#3]
Hi,

This looks like an interesting package, but it's not clear to me how the use case is different from [TupleVectrors](https://github.com/cscherrer/TupleVectors.jl) or [TypedTables](https://github.com/JuliaData/TypedTables.jl). I see that it uses a pointer-based internal representation, I just don't immediately see the implications of that. Could you give some more detail on the design, and where you see this going?

Also, I'm talking with @oschulz and @andyferris in https://github.com/JuliaData/TypedTables.jl/issues/73 about the possibility of merging these two packages, or at least seeing if it can make sense to factor out the commonalities. Of course you're welcome to join if a common effort might be useful to you.

Anyway, I just want to be sure you're aware of all of this in case there might be any redundancy, and to be sure we can guide users toward an appropriate package for a given use case. Thanks :)

**Comment by tkf** [comment]
Thanks for the heads up :) Last time I checked, both TypedTables and StructArrays use the structure-of-arrays (SOA) memory layout. From quickly playing with it, TupleVectors also seem to use SOA.

RecordArrays.jl is different since it uses the array-of-structures (AOS) memory layout. I wrote RecordArrays.jl as a basis of cache-aligned task-local storages like the ones I need for [EventTracker.jl](https://github.com/tkf/EventTracker.jl) and [ConcurrentCollections.jl](https://github.com/tkf/ConcurrentCollections.jl). On the other hand, you'd almost always want to use SOA ("columnar") layout in compute-intensive tasks. So, I don't think it's ideal to combine RecordArrays.jl with other SOA packages. (Though I probably should link to TypedTables.jl and TupleVectors.jl when I mention StructArrays.)
